### PR TITLE
Add monitoring and healthz based on tunnel health.

### PR DESCRIPTION
**Pull request by brendandburns**
@cjcullen @thockin

This is in response to some observed issues in GKE.

### CODE REVIEW

**Comment by brendandburns** [comment]
Closes https://github.com/GoogleCloudPlatform/kubernetes/issues/11251

**Comment by thockin** [comment]
I don't know the SSH logic well, but this changes seems plausible.

**Comment by thockin** [comment]
LGTM, but @cjcullen who maybe has better context

File: pkg/master/master.go [hunk 1]
```diff
@@ -541,8 +550,14 @@ func (m *Master) init(c *Config) {
 		} else {
 			glog.Errorf("Failed to cast %v to HTTPKubeletClient, skipping SSH tunnel.")
 		}
+		healthzChecks = append(healthzChecks, healthz.NamedCheck("SSH Tunnel Check", m.IsTunnelSyncHealthy))
 	}
 
+	m.lastSyncMetric = prometheus.NewGaugeFunc(prometheus.GaugeOpts{
```
Comment (cjcullen) [inline pkg/master/master.go]: Should this be inside the above if block? I don't know prometheus well enough to know the implications, but this metric only makes sense in the ssh-tunnel case.

**Comment by brendandburns** [comment]
Updated to reflect offline conversation with @cjcullen move the metric inside the if ssh-tunnels block.

ptal.

**Comment by brendandburns** [comment]
@thockin @cjcullen can I get an LGTM and an ok-to-merge?

**Comment by cjcullen** [comment]
Sorry, thought I had already.

LGTM

**Comment by erictune** [comment]
I'm not merging any code until Jenkins is green for a turn.

**Pull request merged**